hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -226,6 +226,12 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
       throw new NoSuchTableException("Invalid identifier: %s", from);
     }
 
+    if (!namespaceExists(originalTo.namespace())) {
```

Comment:
While reviewing https://github.com/apache/iceberg/pull/9852 I noticed that there's no such check in Hive that checks whether that namespace exists